### ISSUE TITLE
Missing viewConfig in ForeignKeyPickerData

### ISSUE THREAD

**Issue by gitpm** [OneIdentity/IdentityManager.Imx#183]
**Internal Issue created: 472280**

To my point of view the "ForeignKeyPickerData" - interface -> [Link](https://github.com/OneIdentity/IdentityManager.Imx/blob/v92/imxweb/projects/qbm/src/lib/fk-advanced-picker/foreign-key-picker-data.interface.ts) is missing a  **viewConfig** - Configuration-parameter in order to control the view of the fkSelector.

**Hence this is my suggestion:**

```typescript
export interface ForeignKeyPickerData {
    fkRelations: IForeignKeyInfo[];
    selectedTableName?: string;
    idList?: string[];
    isMultiValue?: boolean;
    isRequired?: boolean;
    disabledIds?: string[];
    viewConfig?: DataSourceToolbarViewConfig;  **// ccc-change**
}
```

in the fk-selector.component.ts I use this parameter to control the viewConfig:
```typescript
 this.settings = {
          dataSource: this.builder.buildReadWriteEntities(
            await this.selectedTable.Get(navigationState),
            this.entitySchema
          ),
          displayedColumns,
          entitySchema: this.entitySchema,
          filters: this.filters,
          dataModel: this.dataModel,
          navigationState,
          filterTree: {
            multiSelect: true,
            filterMethode: async (parentKey) => this.selectedTable.GetFilterTree(parentKey)
          },
          viewConfig: this.data.viewConfig   **// ccc-change**
        };
```
        
  The new parameter gives me the possibilty to inject a viewConfig to the fk-selector.
  In my case I used this code to change the fk-selector's view in the new-request.component.ts
  
  ```typescript
  public async openSidesheet(): Promise<void> {
    const idList = MultiValue.FromString(this.orchestration.recipients.Column.GetValue()).GetValues();
    const response: FKAdvancedPickerResponse = await this.sidesheetService.open(FkAdvancedPickerComponent, {
      title: await this.translateService.get('#LDS#Heading Select Recipients').toPromise(),
      icon: 'user',
      width: 'max(600px, 50%)',
      padding: '0px',
      disableClose: true,
      testId: 'new-requests-recipients-sidesheet',
      data: {
        displayValue: '',
        isRequired: true,
        fkRelations: this.recipientsApi.getFKRelations(),
        isMultiValue: true,
        idList,
        **// ccc-change begin**
        viewConfig:  {
          viewId: 'ccc-view',
            viewConfigs: [{
            ViewId: 'ccc-view',
            IsReadOnly: false,
            UseAsDefault: true,
            AdditionalTableColumns: ['UID_Department',  'CCC_ContractType', 'IdentityType']
          }]
       }
       **// ccc-change-end**
      },
    }).afterClosed().toPromise();`
```

By the way an example how to use the viewConfig and to control the columns,  add optional columns per default, modify the DataModel in the API for a new/given API-Endpoint would be very helpful. 
My impression is that with the PersonConfig-Settings in the QER-API-Code  like e.g.  '.VI_MyData_WhitePages_ResultAttributes' which contains a fixed set of colums. it is very hard and complex to modify these given sets of result columns in the ui-result output

**Comment by Mathnstein** [comment]
Ah this is a cool extension (given I brought the core functionality of view configs in)! I'll see this as a feature extension and mark it down as something that would be nice to have.

**Comment by Mathnstein** [comment]
@gitpm This was a cool extension - I handled it even more generically so that you can put/post/delete configs from the FK selector like you would any other table with this capability.

I've applied it to the newrequest recipients component but we can add this now to many components that make use of the fkselector

![Image](https://github.com/user-attachments/assets/1167a13b-eeee-488e-bad5-42d4c7b707a2)

**Comment by bogdankhamelyuk** [comment]
is this solution implemented in any certain fixes? Unfortunately cant find in 9.3 version.

**Comment by kovtib** [comment]
@bogdankhamelyuk it's only implemented from v10.0.
